cpp/src/arrow/filesystem/azurefs.cc

```diff
@@ -970,6 +970,79 @@ class AzureFileSystem::Impl {
     return stream;
   }
 
+ private:
+  Status DeleteDirContentsWihtoutHierarchicalNamespace(const AzureLocation& location,
+                                                       bool missing_dir_ok) {
+    auto container_client =
+        blob_service_client_->GetBlobContainerClient(location.container);
+    Azure::Storage::Blobs::ListBlobsOptions options;
+    options.Prefix = internal::EnsureTrailingSlash(location.path);
```

Comment:
`EnsureTrailingSlash()` does nothing for empty string:
https://github.com/apache/arrow/blob/d66780d90c6faf7bf051cbf12b4b68dff098bd54/cpp/src/arrow/filesystem/path_util.cc#L151-L159

Should we add a (needless) special condition such as the following for readability?

```cpp
if (!location.path.empty()) {
  options.Prefix = internal::EnsureTrailingSlash(location.path);
}
```

Reply:
Maybe just leave a comment saying that `EnsureTrailingSlash` doesn't alter empty string.

Reply:
Wow. From the name, I though it **ensured** a trailing slash on the output. 🥲

I like this because `Prefix` is a nullable, so not doing anything to it makes even more sense than setting it to a `""` value.

```cpp
if (!location.path.empty()) {
  options.Prefix = internal::EnsureTrailingSlash(location.path);
}
```

Reply:
OK. I'll use the `if (!location.path.empty())`.